packages/in_app_purchase/lib/src/in_app_purchase_connection/in_app_purchase_connection.dart

```diff
@@ -7,15 +7,243 @@ import 'dart:io';
+  /// * If a purchase is not completed([completePurchase] is not called on the purchase object) from the last APP session. Purchase updates will happen when a new APP session starts.
+  ///
+  /// IMPORTANT! To Avoid losing information on purchase updates, You should listen to this stream as soon as your APP launches, preferably before returning your main App Widget in main().
+  /// We recommend to have a single subscription listening to the stream at a given time. If you choose to have multiple subscription at the same time, you should be careful at the fact that each subscription will receive all the events after they start to listen.
+  Stream<List<PurchaseDetails>> get purchaseUpdatedStream => _getStream();
+
+  Stream<List<PurchaseDetails>> _purchaseUpdatedStream;
+
+  Stream<List<PurchaseDetails>> _getStream() {
+    if (_purchaseUpdatedStream != null) {
+      return _purchaseUpdatedStream;
+    }
+
+    if (Platform.isAndroid) {
+      _purchaseUpdatedStream =
+          GooglePlayConnection.instance.purchaseUpdatedStream;
+    } else if (Platform.isIOS) {
+      _purchaseUpdatedStream =
+          AppStoreConnection.instance.purchaseUpdatedStream;
+    } else {
+      throw UnsupportedError(
+          'InAppPurchase plugin only works on Android and iOS.');
+    }
+    return _purchaseUpdatedStream;
+  }
+
   /// Returns true if the payment platform is ready and available.
   Future<bool> isAvailable();
 
   /// Query product details list that match the given set of identifiers.
   Future<ProductDetailsResponse> queryProductDetails(Set<String> identifiers);
 
+  /// Make a payment
+  ///
+  /// This method does not return anything. Instead, after triggering this method, purchase updates will be sent to [purchaseUpdatedStream].
+  /// You should [Stream.listen] to [purchaseUpdatedStream] to get [PurchaseDetails] objects in different [PurchaseDetails.status] and
+  /// update your UI accordingly. When the [PurchaseDetails.status] is [PurchaseStatus.purchased] or [PurchaseStatus.error], you should deliver the content or handle the error, then call
+  /// [completePurchase] to finish the purchasing process.
+  ///
+  ///
+  /// The `productID` is the product ID to create payment for.
+  /// The `applicationUserName`
+  /// The 'sandboxTesting' is only necessary to set to `true` for testing on iOS. The default value is `false`.
+  /// You can find more details on testing payments on iOS [here](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/StoreKitGuide/Chapters/ShowUI.html#//apple_ref/doc/uid/TP40008267-CH3-SW11).
+  /// You can find more details on testing payments on Android [here](https://developer.android.com/google/play/billing/billing_testing).
+  void makePayment(
+      {@required String productID,
+      String applicationUserName,
+      bool sandboxTesting = false});
+
+  /// Completes a purchase either after delivering the content or the purchase is failed. (iOS only).
```

Comment:
In general I'd like this API to be as consistent across platforms as possible. This is a non op on Play but I'd prefer if we asked users to call this in all cases, instead of checking for platform differences. This applies to a few of the API comments below.